### PR TITLE
Exception tables

### PR DESCRIPTION
**Pull request by Jett59**
Adds support for exception handling (aarch64) and interrupt handling (x86). At the moment neither of them do anything fancy, but they give a nice little register dump and make debugging a whole lot more feasible.

### CODE REVIEW

File: kernel/src/aarch64/arch_api/init.rs [hunk 1]
```diff
@@ -87,4 +90,9 @@ pub fn arch_init() {
             },
         });
     }
+
+    // Manually trigger an exception to test the exception handling.
+    unsafe {
+        core::arch::asm!("svc #0");
+    }
```
Comment (luketn) [inline kernel/src/aarch64/arch_api/init.rs:97]: Sure you want to leave in the exception?

File: kernel/src/aarch64/exceptions.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
+use core::{
```
Comment (luketn) [inline kernel/src/aarch64/exceptions.rs:1]: I find this file and the accompanying.s file a little cryptic. Could you include a comment or two explaining what this does, maybe with a link to docs?

File: kernel/src/main.rs [hunk 3]
```diff
@@ -3,7 +3,7 @@
 #![cfg_attr(not(test), no_main)]
 // Lets just hope these aren't as unstable as the language says they are (it would be a pain to have to change everywhere one of these is used)
 #![feature(
-    core_intrinsics,
+    //core_intrinsics,
```
Comment (luketn) [inline kernel/src/main.rs:6]: Is this change intended? If so maybe just delete the dead code?

File: kernel/src/x86_64/arch_api/init.rs [hunk 4]
```diff
@@ -24,4 +27,9 @@ pub fn arch_init() {
         ),
     );
     paging::initialize_paging();
+
+    // Trigger an exception
+    unsafe {
+        core::arch::asm!("mov rax, 0xdeadbeef", "int 0x80");
+    }
```
Comment (luketn) [inline kernel/src/x86_64/arch_api/init.rs:34]: Shall we remove this?

**Review by luketn — APPROVED** [review]
Looks great! Incredible job. I have run the aarch64 version locally and it works well.

File: kernel/src/aarch64/arch_api/init.rs (hunk 1 above)
Comment (Jett59) [reply kernel/src/aarch64/arch_api/init.rs:97]: I'll remove them both when I've made sure everything is working properly. I just didn't want to take them out before finishing the changes.
